app/views/govuk_publishing_components/components/_input.html.erb

```diff
@@ -63,14 +64,19 @@
     type = "text"
     inputmode = "numeric"
   end
+
+  dir = right_to_left ? "rtl" : nil
 %>
 
 <%= content_tag :div, class: form_group_css_classes do %>
   <% if label %>
+    <% label_rtl = label[:right_to_left] == false ? label[:right_to_left] : right_to_left %>
```

Comment:
I think I understand this logic - if the label rtl is set to false, then use that, otherwise use the 'general' right_to_left value, is that right? It might be fractionally easier to read as shown below, but I notice that this logic repeats for the hint and error - we might have reached the tipping point where the amount of logic in the view justifies putting some of this code into a helper file, where code comments and a dedicated helper test would help clarify the functionality.

```
label_rtl = label[:right_to_left] == false ? false : right_to_left
```

Reply:
Since this functionality is being applied to more than one component and could apply to others in future, it might be something to add to the shared helper.

Reply:
This is now outdated and improved by the later changes along the lines suggested by Kevin.